CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+# Contributing to Yarn
+
+Yarm is in a slightly precarious position, due to the odd nature of MCP. MCP is all-rights-reserved, yet visible to anyone
+who has ever modded Minecraft using anything other than Fabric. This leads to "contamination" of a lot of potential 
+contributors, since people already have a conception of what a method "should" be named. Because of this, we have decided to 
+take a clean-room approach with mapping Minecraft to avoid potential legal/drama issues within the community.
+
+## What's in a Name?
+
+The fundamental goal of mappings is to make modding Minecraft easier to understand, for beginners and experts alike. We want
+to create a cohesive, simple naming scheme that avoids confusion or being too verbose. Having a lot of things named in
+different schemes would be a nightmare for usability, so we want to avoid that. This includes naming schemes not just within
+Yarn but within the greater Minecraft community. Despite that goal, there are some name sources that are inherently more
+"reliable" than others; you should use an up-to-date and known Mojang name above any alternative, even if an alternative
+would be simpler.
+
+## Guidelines
+
+To keep with the goal of consistency within the mappings and the greater Minecraft community, here are a set of guidelines to
+follow when naming unmapped methods.
+
+### 1. Use modern Mojang names when known.
+
+This can be a contentious decision, due to many of Notch's original names for things (that he himself has stated were
+terrible choices for names) sticking around. While it would be nice to have names like `Gui` or the like, we have decided to 
+use `Screen`, since that's the name for guis that the client-side registry uses. Mojang names can be found in many places,
+from registry names to lang files to tweets by devs. If you want to claim something as a Mojang name, and it's not an obvious
+one, provide a source for proof.
+
+Minecraft is still under active development, and names have been chaned in the past. The most notable case of this is `Tile`,
+which used to be the name for Blocks. This has changed in 1.13 and 1.14, so we use the modern name `BlockEntity` instead of
+the old Notch name `TileEntity`.
+
+### 2. Avoid MCP names wherever possible.
```

Comment:
Though not being a real situation, I would word this title defensively - that said, I would say "Avoid MCP names and other proprietary names wherever possible".